keylime/ima/ima.py

```
@@ -466,160 +406,92 @@ def read_allowlist(alist=None, checksum="", al_sig_file=None, al_key_file=None):
-
-        alist = copy.deepcopy(EMPTY_ALLOWLIST)
-        alist["meta"]["timestamp"] = str(datetime.datetime.now())
-        alist["meta"]["generator"] = "keylime-legacy-format-upgrade"
-        if checksum:
-            alist["meta"]["checksum"] = checksum
-
-        for line in alist_raw.splitlines():
-            line = line.strip()
-            if len(line) == 0:
-                continue
-
-            pieces = line.split(None, 1)
-            if not len(pieces) == 2:
-                logger.warning("Line in AllowList does not consist of hash and file path: %s", line)
-                continue
-
-            (checksum_hash, path) = pieces
-
-            if path.startswith("%keyring:"):
-                entrytype = "keyrings"
-                path = path[len("%keyring:") :]  # remove leading '%keyring:' from path to get keyring name
-            else:
-                entrytype = "hashes"
-
-            if path in alist[entrytype]:
-                alist[entrytype][path].append(checksum_hash)
-            else:
-                alist[entrytype][path] = [checksum_hash]
-
-    alist = update_allowlist(alist)
+        if not signing.verify_signature(ima_policy_key, ima_policy_sig, ima_policy):
+            raise ImaValidationError(message="IMA policy failed detached signature verification!", code=401)
+        logger.info("IMA policy passed detached signature verification")
 
-    return alist
-
-
-def read_excllist(exclude_path=None):
-    excl_list = []
-    if exclude_path and os.path.exists(exclude_path):
-        with open(exclude_path, encoding="utf-8") as f:
-            for line in f:
-                line = line.strip()
-                if line.startswith("#") or len(line) == 0:
-                    continue
-                excl_list.append(line)
-
-        logger.debug("Loaded exclusion list from %s: %s", exclude_path, excl_list)
+    # validate that the allowlist is proper JSON
+    try:
+        lists = json.loads(ima_policy)
```

Do we want to check against schema also?

---

I think that would be a good idea, though, like you mentioned, we'd need to investigate the state of dependencies for that functionality before implementing.

In the meantime, I've extended this policy correctness check to also apply to the Keylime tenant (so whatever checks are present fail early).

---

Makes sense. Schema validation can be easily implemented after this PR is merged.